sdk/appconfiguration/app-configuration/src/featureFlag.ts

```diff
@@ -0,0 +1,345 @@
+   */
+  name: "Microsoft.TimeWindow";
+  /**
+   * Parameters that can be passed in the filter.
+   */
+  parameters: {
+    start: string;
+    end: string;
+  };
+}
+
+/**
+ * Percentage Client filter for the feature flag configuration setting.
+ *
+ * [More Info](https://docs.microsoft.com/en-us/azure/azure-app-configuration/howto-feature-filters-aspnet-core)
+ */
+export interface FeatureFlagPercentageClientFilter {
+  /**
+   * The name of the feature filter.
+   */
+  name: "Microsoft.Percentage";
+  /**
+   * Parameters that can be passed in the filter.
+   * Value is expected to be from 0 to 100. SDK doesn't validate the value.
+   */
+  parameters: {
+    value: number;
+  };
+}
+
+/**
+ * FeatureFlag represents a configuration setting that stores a feature flag value.
+ * [More Info](https://docs.microsoft.com/azure/azure-app-configuration/concept-feature-management)
+ *
+ * This helper method tells you if the given setting is a feature flag.
+ */
+export function isFeatureFlag(setting: ConfigurationSetting | FeatureFlag): setting is FeatureFlag {
+  return setting.contentType === featureFlagContentType;
+}
+
+/**
+ * This helper method tells you if the given client filter has the name "Microsoft.Targeting".
+ */
+function isFeatureFlagTargetingClientFilter(
+  clientFilter: any
+): clientFilter is FeatureFlagTargetingClientFilter {
+  return clientFilter.name === "Microsoft.Targeting";
+}
+
+/**
+ * This helper method tells you if the given client filter has the name "Microsoft.TimeWindow".
+ */
+function isFeatureFlagTimeWindowClientFilter(
+  clientFilter: any
+): clientFilter is FeatureFlagTimeWindowClientFilter {
+  return clientFilter.name === "Microsoft.TimeWindow";
+}
+
+/**
+ * This helper method tells you if the given client filter has the name "Microsoft.Percentage".
+ */
+function isFeatureFlagPercentageClientFilter(
+  clientFilter: any
+): clientFilter is FeatureFlagPercentageClientFilter {
+  return clientFilter.name === "Microsoft.Percentage";
+}
+
+/**
+ * Type of FeatureFlag based on the client filters.
+ */
+export type FeatureFlagType<
+  T extends "targeting" | "timeWindow" | "percentage"
+> = T extends "targeting"
+  ? FeatureFlagTargetingClientFilter
+  : T extends "timeWindow"
+  ? FeatureFlagTimeWindowClientFilter
+  : T extends "percentage"
+  ? FeatureFlagPercentageClientFilter
+  : never;
+
+/**
+ * This helper method tells you if the given client filter is a FeatureFlagClientFilter.
+ */
+export function isFeatureFlagClientFilter<T extends "targeting" | "timeWindow" | "percentage">(
+  type: T,
+  obj: any
+): obj is FeatureFlagType<T> {
+  switch (type) {
+    case "targeting":
+      return isFeatureFlagTargetingClientFilter(obj);
+    case "timeWindow":
+      return isFeatureFlagTimeWindowClientFilter(obj);
+    case "percentage":
+      return isFeatureFlagPercentageClientFilter(obj);
+    default:
+      throw new Error(`Invalid feature flag filter type ${type}`);
```

Comment:
I might have written it this way _but_ we should probaby just return 'false' here since it's a typeguard.

Reply:
(also, this preps you to just return it verbatim, so the user still has the data in case the number of flags expands in the future)